.github/workflows/release.yml

```diff
@@ -0,0 +1,67 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - master
+jobs:
+  release:
+    name: Generate Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.RELEASEBOT_TOKEN }}
```

Comment:
Can't you use the current user's token?

Reply:
We cannot. Since we use branch protection, the user needs permission to commit on master without an approval in order to perform the release function. The default github token will not have this permission, but the release bot does.
This should be relatively safe, as the secrets are not provided to forks of the repo, and only trusted users are allowed to create branches in the primary repo.

Reply:
I was thinking from a fork standpoint, whether people who fork can set these variables to reasonable values that will make it work for them...

So you are saying even if the user has permission to perform, the default token github makes available doesn't have this "scope" in it? or that contributors of this repo won't? if it's the latter, then that's ok because the person who clicked "Squash and Merge" has the needed permissions to merge into master, so their token will have the same permissions. If it's the former, then sure, I understand.

Reply:
Ah yeah so this isn't set up to allow forks to publish releases.
I'm not certain that the action runs as the user who merges. I think it runs as the `github-actions` account? I _think_ that account does have write access but is not considered an Admin (and thus can't push directly to master). I'd prefer to merge this as-is for now and then explore ways to make it compatible with forks. It may require having all of the git credentials specified as repository secrets to get it working correctly.